packages/web-components/fast-components/src/design-tokens.ts

```diff
@@ -294,12 +294,145 @@ export const accentPalette = create<Palette>({
+        ),
+});
+
+/** @public */
+export const neutralLayer1 = create<Swatch>(
+    "neutral-layer-1"
+).withDefault((element: HTMLElement) =>
+    neutralLayer1Recipe.getValueFor(element).evaluate(element)
+);
+
+// Neutral Layer 2
+/** @public */
+export const neutralLayer2Recipe = create<ColorRecipe>({
+    name: "neutral-layer-2-recipe",
+    cssCustomPropertyName: null,
+}).withDefault({
+    evaluate: (element: HTMLElement): Swatch =>
+        neutralLayer2Algorithm(
+            neutralPalette.getValueFor(element),
+            baseLayerLuminance.getValueFor(element),
+            neutralFillLayerRestDelta.getValueFor(element),
+            neutralFillRestDelta.getValueFor(element),
+            neutralFillHoverDelta.getValueFor(element),
+            neutralFillActiveDelta.getValueFor(element)
+        ),
+});
+
+/** @public */
+export const neutralLayer2 = create<Swatch>(
+    "neutral-layer-2"
+).withDefault((element: HTMLElement) =>
+    neutralLayer2Recipe.getValueFor(element).evaluate(element)
+);
+
+// Neutral Layer 3
 /** @public */
-export const fillColor = create<Swatch>("fill-color").withDefault(element => {
-    const palette = neutralPalette.getValueFor(element);
-    return palette.get(palette.swatches.length - 5);
+export const neutralLayer3Recipe = create<ColorRecipe>({
+    name: "neutral-layer-3-recipe",
+    cssCustomPropertyName: null,
+}).withDefault({
+    evaluate: (element: HTMLElement): Swatch =>
+        neutralLayer3Algorithm(
+            neutralPalette.getValueFor(element),
+            baseLayerLuminance.getValueFor(element),
+            neutralFillLayerRestDelta.getValueFor(element),
+            neutralFillRestDelta.getValueFor(element),
+            neutralFillHoverDelta.getValueFor(element),
+            neutralFillActiveDelta.getValueFor(element)
+        ),
 });
 
+/** @public */
+export const neutralLayer3 = create<Swatch>(
+    "neutral-layer-3"
+).withDefault((element: HTMLElement) =>
+    neutralLayer3Recipe.getValueFor(element).evaluate(element)
+);
+
+// Neutral Layer 4
+/** @public */
+export const neutralLayer4Recipe = create<ColorRecipe>({
+    name: "neutral-layer-4-recipe",
+    cssCustomPropertyName: null,
+}).withDefault({
+    evaluate: (element: HTMLElement): Swatch =>
+        neutralLayer4Algorithm(
+            neutralPalette.getValueFor(element),
+            baseLayerLuminance.getValueFor(element),
+            neutralFillLayerRestDelta.getValueFor(element),
+            neutralFillRestDelta.getValueFor(element),
+            neutralFillHoverDelta.getValueFor(element),
+            neutralFillActiveDelta.getValueFor(element)
+        ),
+});
+
+/** @public */
+export const neutralLayer4 = create<Swatch>(
+    "neutral-layer-4"
+).withDefault((element: HTMLElement) =>
+    neutralLayer4Recipe.getValueFor(element).evaluate(element)
+);
+
+/** @public */
+export const fillColor = create<Swatch>("fill-color").withDefault(element =>
+    neutralLayer1.getValueFor(element)
```

Comment:
Should fill color be dependent on the idea of a *current layer*? Or is fill color always based on neutralLayer1?

Reply:
It's not always layer 1. Under the current model it's up to someone to manually set the correct fill for the given element though. I think we structured it this way to make it easier to piece things together, for instance to build a nav component that's always "layer 2" and a toolbar that's "layer 3" and allow you to compose your app structurally however makes sense rather than force nesting and infer layer boundaries. I think that's what you were asking anyway.

So someone might do `fillColor.setValueFor(navContainer, neutralLayer2)` or the same within a component similar for how we do Card.

Reply:
I think what I was wondering is should we have `fillColor` be something like this?

```ts
export const fillColor = create<Swatch>("fill-color").withDefault(element => {
    const currentLayer = layer.getValueFor(element);
    return currentLayer.getValueFor(element);
});
```

This would introduce a token that you can set to indicate what layer a DOM tree is in, and then base fill off of that.

Reply:
Ah, that's interesting. Would `layer` be the color recipe `neutralLayer1` or a mapping like `"1"`?

I want to get @nicholasrice thoughts on this as well, but it seems most beneficial to me if we have other tokens that might hang off `layer`, like perhaps that's how someone could apply different elevation values automatically (a "Floating" layer uses elevation 32, etc.)

Although, as I started writing that example, that felt very similar to the style-set model we've been considering in the modular styling approach. For instance, an "accent plate" might be a set consisting of "accent fill", "neutral border", and "foreground on accent" for content.

That seems like a great model though, keeping sets as tokens. I'm not sure the structure this should take right now, but I think it's a good consideration. I suggest we use the model I have in code for now since it works and it's less abstracted, but continue with thoughts on the evolution of compound or dependent styling.

Reply:
Rather than having it return a number, I was thinking it returned a token. It's a token that specifies which token to use. I have no idea if this is a good idea technically. Nick would definitely need to weigh in.

Reply:
I think whether we did this or not should depend on if we need to coordinate other tokens to the layer or not. If so, then I think creating a layer token to communicate the contextual layer makes a lot of sense, but otherwise it's probably an un-needed abstraction.